common/src/main/java/com/microsoft/identity/common/SharedPreferencesHelper.java

```diff
@@ -0,0 +1,90 @@
+ */
+public class SharedPreferencesHelper {
+
+    private final SharedPreferences mSharedPreferences;
+
+    /**
+     * Constructs an instance of SharedPreferencesHelper.
+     * The default operating mode is {@link Context#MODE_PRIVATE}
+     *
+     * @param context Interface to global information about an application environment.
+     * @param name    The desired {@link android.content.SharedPreferences} file. It will be created
+     *                if it does not exist.
+     */
+    public SharedPreferencesHelper(
+            final Context context,
+            final String name) {
+        mSharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE);
+    }
+
+    /**
+     * Constructs an instance of SharedPreferencesHelper.
+     *
+     * @param context       Interface to global information about an application enviroment.
+     * @param name          The desired {@link SharedPreferences} file. It will be created
+     *                      if it does not exist.
+     * @param operatingMode Operating mode {@link Context#getSharedPreferences(String, int)}.
+     */
+    public SharedPreferencesHelper(
+            final Context context,
+            final String name,
+            final int operatingMode) {
+        mSharedPreferences = context.getSharedPreferences(name, operatingMode);
+    }
+
+    /**
+     * Saves a Token (as a {@link String} to the {@link SharedPreferences} file.
+     *
+     * @param key   The name (key) of the Token to save.
+     * @param value The Token's value (as a {@link String}).
+     * @return True, if the value was successfully written.
+     */
+    public final boolean saveToken(
+            final String key,
+            final String value) {
+        final SharedPreferences.Editor editor = mSharedPreferences.edit();
+        editor.putString(key, value);
+        return editor.commit();
+    }
+
+    /**
+     * Asynchronously saves a Token (as a {@link String} to the {@link SharedPreferences} file.
+     *
+     * @param key   The name (key) of the Token to save.
+     * @param value The Token's value (as a {@link String}).
+     */
+    public final void saveTokenAsync(
+            final String key,
+            final String value) {
+        final SharedPreferences.Editor editor = mSharedPreferences.edit();
+        editor.putString(key, value);
+        editor.apply();
+    }
+
+    /**
+     * Retrieves a Token from the {@link SharedPreferences} file.
+     *
+     * @param key The name (key) of the Token.
+     * @return The Token's value or null if no value could be found.
+     */
+    public final String getToken(final String key) {
+        return mSharedPreferences.getString(key, null);
+    }
+
+    /**
+     * Gets the underlying {@link SharedPreferences} instance.
+     *
+     * @return The SharedPreference instance used by this SharedPreferencesHelper.
+     */
+    public final SharedPreferences getSharedPreferences() {
```

Comment:
Why did you want to expose the underlying SharedPreferences object?

Reply:
My thinking was that I'd expose the underlying instance because this class is `public` and doesn't provide anything like complete parity to the `SharedPreferences` API -- Come to think of it, I think I'll yank this out and  add a `contains()`, `clear()`, and `remove(String)` method to cover other likely-necessary features